ts/features/walletV3/payment/hooks/usePagoPaPayment.ts

```diff
@@ -0,0 +1,113 @@
+
+type PagoPaPaymentParams = Omit<PaymentInitStateParams, "startRoute">;
+
+/**
+ * A hook for initiating a PagoPA payment flow.
+ * This hook provides functions to start a payment flow using various input methods.
+ * @returns An object containing functions to start different types of payment flows.
+ */
+const usePagoPaPayment = () => {
+  const route = useRoute();
+  const dispatch = useIODispatch();
+  const navigation = useIONavigation();
+
+  /**
+   * Initializes the payment state based on the provided parameters.
+   * The initialization includes the store of the current route which allows the app to
+   * return to it when the payment flow is finished.
+   * @param {PagoPaPaymentParams} params - Parameters for initializing the payment state.
+   */
+  const initPaymentState = ({ startOrigin }: PagoPaPaymentParams) => {
+    const startRoute: PaymentStartRoute = {
+      routeName: route.name as keyof AppParamsList,
+      routeKey:
+        route.key as keyof NavigatorScreenParams<AppParamsList>["screen"]
+    };
+    dispatch(
+      walletPaymentInitState({
+        startRoute,
+        startOrigin
+      })
+    );
+  };
+
+  /**
+   * Initiates the payment flow using the provided RptId string and additional parameters.
+   * @param {RptId} rptId - The RptId for the payment flow.
+   * @param {PagoPaPaymentParams} params - Additional parameters for the payment flow.
+   */
+  const startPaymentFlow = (rptId: RptId, params: PagoPaPaymentParams = {}) => {
+    initPaymentState(params);
+    navigation.navigate(WalletPaymentRoutes.WALLET_PAYMENT_MAIN, {
+      screen: WalletPaymentRoutes.WALLET_PAYMENT_DETAIL,
+      params: {
+        rptId
+      }
+    });
+  };
+
+  /**
+   * Initiates the payment flow using the provided PagoPA RptId and additional parameters.
+   * @param {PagoPaRptId} rptId - The PagoPA RptId for the payment flow.
+   * @param {PagoPaPaymentParams} params - Additional parameters for the payment flow.
+   */
+  const startPaymentFlowWithRptId = (
+    rptId: PagoPaRptId,
+    params: PagoPaPaymentParams = {}
+  ) => {
+    pipe(
+      O.fromNullable(rptId),
+      O.map(PagoPaRptIdFromString.encode),
+      O.map(RptId.decode),
+      O.chain(O.fromEither),
+      O.map(rptIdString => startPaymentFlow(rptIdString, params))
+    );
+  };
+
+  /**
+   * Initiates the payment flow using the provided payment data and additional parameters.
+   * @param {Object} data - Payment data containing the payment notice number and an organization fiscal code.
+   * @param {PagoPaPaymentParams} params - Additional parameters for the payment flow.
+   */
+  const startPaymentFlowWithData = (
+    data: {
+      paymentNoticeNumber: string;
+      organizationFiscalCode: string;
+    },
+    params: PagoPaPaymentParams = {}
+  ) => {
+    pipe(
+      PagoPaRptIdFromString.decode(data),
+      E.chain(RptId.decode),
+      E.map(rptIdString => startPaymentFlow(rptIdString, params))
+    );
```

Comment:
Here it hangs on the first decode, with a `readablePrivacyReport` of ` some value at [root] is not a valid [RptIdFromString]`.

I've tried using both the random RptId and data I found around our docs, but to no avail.

The only fix I found useful is using the same code that was used in the `WalletPaymentPlayground`, but I'm pretty sure there is a reason for the double decode.

It is weird, though, that it runs just fine on your machine; I have tried reinstalling and regenerating all types and dependencies, but the situation stays the same on my machine: are you sure that you don't have some edited definitions or forgot to push something?